consul/catalog_endpoint.go

```diff
@@ -36,6 +37,17 @@ func (c *Catalog) Register(args *structs.RegisterRequest, reply *struct{}) error
 			return fmt.Errorf("Must provide service name with ID")
 		}
 
+		// Verify that the service name is valid per RFC-1123 and
+		// RFC-2782.
+		//
+		// FIXME(sean@): After 0.7 has been branched, change from a
+		// warning to an error for the 0.8 release.
+		re := regexp.MustCompile(`^(?i:[a-z0-9]|[a-z0-9][a-z0-9\-]{0,61}[a-z0-9])$`)
```

Comment:
It should be a package-global variable probably? See for precedent:

https://github.com/hashicorp/consul/blob/master/consul/state/prepared_query.go#L13
https://github.com/hashicorp/consul/blob/master/command/agent/command.go#L33
https://github.com/hashicorp/consul/blob/master/command/agent/agent.go#L47